sdk/java/pulumi/src/main/java/com/pulumi/experimental/automation/serialization/internal/ProjectSettingsYamlSerializer.java

```diff
@@ -0,0 +1,386 @@
+        public String default_;
+        public boolean secret;
+
+        @Override
+        public ProjectTemplateConfigValue convert() {
+            return new ProjectTemplateConfigValue(description, default_, secret);
+        }
+    }
+
+    public static class ProjectBackendModel implements Model<ProjectBackend> {
+        public String url;
+
+        @Override
+        public ProjectBackend convert() {
+            return ProjectBackend.builder()
+                    .url(url)
+                    .build();
+        }
+    }
+
+    public static class ProjectOptionsModel implements Model<ProjectOptions> {
+        public String refresh;
+
+        @Override
+        public ProjectOptions convert() {
+            return ProjectOptions.builder()
+                    .refresh(refresh)
+                    .build();
+        }
+    }
+
+    public static class ProjectPluginsModel implements Model<ProjectPlugins> {
+        public List<ProjectPluginOptionsModel> providers;
+        public List<ProjectPluginOptionsModel> languages;
+        public List<ProjectPluginOptionsModel> analyzers;
+
+        @Override
+        public ProjectPlugins convert() {
+            return ProjectPlugins.builder()
+                    .providers(convertList(providers))
+                    .languages(convertList(languages))
+                    .analyzers(convertList(analyzers))
+                    .build();
+        }
+
+        private List<ProjectPluginOptions> convertList(List<ProjectPluginOptionsModel> plugins) {
+            List<ProjectPluginOptions> result = null;
+            if (plugins != null) {
+                result = new ArrayList<>();
+                for (var plugin : plugins) {
+                    result.add(plugin.convert());
+                }
+            }
+            return result;
+        }
+    }
+
+    public static class ProjectPluginOptionsModel implements Model<ProjectPluginOptions> {
+        public String name;
+        public String version;
+        public String path;
+
+        @Override
+        public ProjectPluginOptions convert() {
+            return ProjectPluginOptions.builder()
+                    .name(name)
+                    .version(version)
+                    .path(path)
+                    .build();
+        }
+    }
+
+    private static class ProjectSettingsModelConstructor extends Constructor {
+        public ProjectSettingsModelConstructor(LoaderOptions loaderOptions) {
+            super(ProjectSettingsModel.class, loaderOptions);
+            setPropertyUtils(new PropertyUtils() {
+                @Override
+                public Property getProperty(Class<?> type, String name) {
+                    // First look for the property with the exact name.
+                    var property = super.getProperty(type, name);
+                    if (property instanceof MissingProperty) {
+                        // If the property is missing, try with an underscore at the end.
```

Comment:
Ah the `default_` etc. makes sense now.

Reply:
Added a comment to these fields to explain.